src/server/pfs/fuse/server.go

```diff
@@ -577,22 +592,12 @@ func Server(sopts *ServerOptions, existingClient *client.APIClient) error {
 			http.Error(w, errMsg, webCode)
 			return
 		}
-
+		// Reset DatumState in case datums were mounted
+		mm.resetDatumState()
```

Comment:
Good question. There are potential issues with placing this line on either side of the unmount call. I chose to put it before because the mount server determines if it's in "datum mode" based on datum state. If we reset datum state after unmounting and an error occurs during unmounting, the mount server will still think it's in "datum mode". That shouldn't be the case in this handler, because `/unmount_all` is called when the user backs out of the Test view to the Explore view. 

For the other place you mentioned, I can move resetting datum state to after unmounting. It might make more sense after in that case because we're in the `/datums/_mount` handler.